apicula.dockerfile

```diff
@@ -0,0 +1,46 @@
+# Authors:
+#   Unai Martinez-Corral
+#   Lucas Teske
+#
+# Copyright 2019-2021 Unai Martinez-Corral <unai.martinezcorral@ehu.eus>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+FROM hdlc/build:build AS build
+
+RUN apt-get update -qq \
+ && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends wget
+
+RUN mkdir /opt/apicula \
+ && wget https://files.pythonhosted.org/packages/1a/d6/b3162f87ff114d639095fe7c0655080ee16caff9037d6629f738d8b28d92/Apycula-0.0.1a6.tar.gz \
+ && tar -xvf Apycula-0.0.1a6.tar.gz -C /opt/apicula \
+ && mv /opt/apicula/Apycula-0.0.1a6/* /opt/apicula \
+ && rm -fr /opt/apicula/Apycula-0.0.1a6/ \
+ && ls -lah /opt/apicula
+
+#---
+
+FROM scratch AS pkg
+COPY --from=build /opt/apicula /apicula
```

Comment:
This seems not to be working as expected. See https://github.com/hdl/containers/pull/11/checks?check_run_id=1843399217#step:6:51. `/apicula` is empty in `hdlc/pkg:apicula`.

Reply:
That's weird. How does this test lists the stuff? Since it is working because of the next step uses the same command to copy everything.

Reply:
The test procedure is defined in `dockerTestPkg`: https://github.com/hdl/containers/blob/main/.github/bin/dockerTestPkg#L31-L39.

An image based on `alpine` is created and the content of the pkg is copied:

```
COPY --from=hdlc/pkg:${_tag} /${_dir} /
```

Then, the test script is executed inside that container:

```
docker run --rm -v /$(pwd)://wrk hdlc/testpkg:"${_tag}" //wrk/test/"${_tag}".pkg.sh
```

Since the call in the workflow (above) is `dockerTestPkg apicula`, the specific commands in this case are the following:

```
docker build -t hdlc/testpkg:apicula - <<-EOF
FROM alpine
RUN apk add -U --no-cache file tree
COPY --from=hdlc/pkg:apicula /apicula /
EOF

docker run --rm -v /$(pwd)://wrk hdlc/testpkg:apicula //wrk/test/apicula.pkg.sh
```

---

The `apicula.pkg.sh` script calls `_tree.sh`, which executes `tree /usr/local`.

Therefore, the issue is that you are not placing final usable artifacts in `/apicula/usr/local` inside `hdlc/pkg:apicula`. Instead, you are copying the content of a package/tarball into `/apicula/*`.

It works in the next step, because you then use `python3 setup.py install` for placing the artifacts in the desired locations.